packages/lisk-api-client/src/api_client.js

```diff
@@ -107,9 +114,7 @@ export default class APIClient {
 			{},
 			commonHeaders,
 			options.nethash ? { nethash: options.nethash } : {},
-			{
-				'User-Agent': getUserAgent(options.client),
-			},
+			getUserAgent(options.client),
```

Comment:
This `Object.assign` call combines various headers objects. User agent is on the wrong semantic level for this. The function as written doesn't just get the user agent, it gets a headers object which happens to only include the user agent (currently). We should either use the previous pattern or rename the function something like `getClientHeaders`.